### PR TITLE
fixed a lot of the issues with voice channel perms

### PR DESCRIPTION
**Pull request by DetroitWhiskey136**
**Please describe the changes this PR makes and why it should be merged:**

**Status**

- [X] Code changes have been tested against the Discord API and the discord.js wrapper, or there are no code changes
- [X] Documentation has been added/modified, or there is nothing to change (docs/mods.json)

**Semantic versioning classification:**

- [ ] This PR changes DBM's interface (methods or parameters added to default methods)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: actions/set_voice_channel_permissions_MOD.js [hunk 1]
```diff
@@ -9,80 +9,66 @@ module.exports = {
...
-    const index = parseInt(data.storage, 10);
-    return index < 3 ? `${names[index]}` : `${names[index]} - ${data.varName}`;
+  subtitle(data, presets) {
+    return `${presets.getVoiceChannelText(data.storage, data.varName)}`;
   },
 
   fields: ['storage', 'varName', 'permission', 'state'],
 
   html(isEvent, data) {
+    console.log(data);
```
Comment (almeidx) [inline actions/set_voice_channel_permissions_MOD.js:19]:
```suggestion
```

File: actions/set_voice_channel_permissions_MOD.js [hunk 2]
```diff
@@ -9,80 +9,66 @@ module.exports = {
...
-          this.callNextAction(cache);
-        })
-        .catch(this.displayError.bind(this, data, cache));
-    }
+    options[data.permission] = data.state === '0' ? true : data.state === '1' ? false : null;
+    channel.permissionOverwrites
+      // providing the server id is the same as providing the everyone role id;
+      .edit(server.id, options, { type: 0 })
+      .catch(this.displayError.bind(this, data, cache));
     this.callNextAction(cache);
```
Comment (almeidx) [inline actions/set_voice_channel_permissions_MOD.js:72]:
```suggestion
    channel.permissionOverwrites
      // providing the server id is the same as providing the everyone role id;
      .edit(server.id, options, { type: 0 })
      .then(() => this.callNextAction(cache))
      .catch((err) => this.displayError(data, cache, err));
```

File: actions/set_role_voice_channel_perms_MOD.js [hunk 3]
```diff
@@ -9,96 +9,72 @@ module.exports = {
...
-          .then(() => {
-            this.callNextAction(cache);
-          })
-          .catch(this.displayError.bind(this, data, cache));
-      }
+    if (role?.id) {
+      channel.permissionOverwrites
+        .edit(role.id, options, { type: 0 })
+        .then(() => this.callNextAction(cache))
+        .catch((err) => this.displayError.bind(data, cache, err));
```
Comment (almeidx) [inline actions/set_role_voice_channel_perms_MOD.js:74]:
```suggestion
        .catch((err) => this.displayError(data, cache, err));
```

File: actions/set_voice_channel_permissions_MOD.js [hunk 4]
```diff
@@ -9,81 +9,66 @@ module.exports = {
...
-        })
-        .catch(this.displayError.bind(this, data, cache));
-    }
-    this.callNextAction(cache);
+    options[data.permission] = data.state === '0' ? true : data.state === '1' ? false : null;
+    channel.permissionOverwrites
+      // providing the server id is the same as providing the everyone role id;
+      .edit(server.id, options, { type: 0 })
+      .then(() => this.callNextAction(cache))
+      .catch((err) => this.displayError.bind(data, cache, err));
```
Comment (almeidx) [inline actions/set_voice_channel_permissions_MOD.js:71]:
```suggestion
      .catch((err) => this.displayError(data, cache, err));
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Comment by DetroitWhiskey136** [comment]
i'm so good at my job :)

File: actions/set_voice_channel_permissions_MOD.js [hunk 5]
```diff
@@ -9,81 +9,66 @@ module.exports = {
...
-    } else if (channel && channel.updateOverwrite) {
-      channel
-        .updateOverwrite(server.id, options)
-        .then(() => {
-          this.callNextAction(cache);
-        })
-        .catch(this.displayError.bind(this, data, cache));
-    }
-    this.callNextAction(cache);
+    options[data.permission] = data.state === '0' ? true : data.state === '1' ? false : null;
```
[the remark this answers is not in the document]
Comment (DetroitWhiskey136) [reply actions/set_voice_channel_permissions_MOD.js:66]: idk why my changes didn't take... how strange

[the remark this answers is not in the document]
Comment (DetroitWhiskey136) [reply actions/set_voice_channel_permissions_MOD.js:66]: that array also wasn't supposed to be there

**Comment by DetroitWhiskey136** [comment]
would help if i was on the right branch lol

**Comment by DetroitWhiskey136** [comment]
ok that should have done the right thing this time.

**Review by almeidx — APPROVED** [review]

**Pull request merged**